src/main/java/me/kavin/piped/utils/ResponseHelper.java

```diff
@@ -739,14 +785,20 @@ public static byte[] subscribeResponse(String session, String channelId)
     public static byte[] unsubscribeResponse(String session, String channelId)
             throws IOException {
 
-        User user = DatabaseHelper.getUserFromSession(session);
+        User user = DatabaseHelper.getUserFromSessionWithSubscribed(session);
 
         if (user != null) {
             try (Session s = DatabaseSessionFactory.createSession()) {
-                s.getTransaction().begin();
-                s.createNativeQuery("delete from users_subscribed where subscriber = :id and channel = :channel")
-                        .setParameter("id", user.getId()).setParameter("channel", channelId).executeUpdate();
-                s.getTransaction().commit();
+                if (user.getSubscribed().contains(channelId)) {
+                    Set<String> subscribed = user.getSubscribed();
+                    subscribed.removeIf(sub -> sub.equals(channelId));
+                    user.setSubscribed(subscribed);
+
+                    s.update(user);
+                    s.getTransaction().begin();
+                    s.getTransaction().commit();
+                }
+

```

Comment:
What's wrong with the existing code?

Reply:
After some research, it does seem to protect against SQL injections. @FireMasterK do you want me to change it back?

Reply:
I've just reverted it :+1: